src/Plugins/Devices/DeviceProxy_LibUSB.cpp

```diff
@@ -441,9 +442,16 @@ void DeviceProxy_LibUSB::receive_data(uint8_t endpoint, uint8_t attributes, uint
 
 	int rc = LIBUSB_SUCCESS;
 
-	if (timeout < 10)
-		timeout = 10;	//TODO: explain this!
+	//if (timeout < 10)
+		//timeout = 10;	//TODO: explain this!
 
+	/* Skazza 20-10-2016
+	 * Infinite timeout to avoid stalling with mass storages, temporarily assigned here.
+	 * Need to check if this is useful or only do-while is needed. But for now it works.
+	 */
+	timeout = 0; 
```

Comment:
To make the code a bit smarter, we can set the timeout to 0 only if the device is mass storage (so with bInterfaceClass = 0x08), in order to avoid strange behaviours with other devices. What do you think?